### PR TITLE
AB#7310 Add the quick checkout page

### PR DESCRIPTION
**Pull request by crhistianramirez**

### CODE REVIEW

File: Website/src/rendering/src/edge/ordercloud/responses.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  return errorResponse(response, 'Invalid Client', 'Authorization.InvalidClient', 403);
+}
+
+export function errorResponse(
+  response: NextApiResponse,
+  message: string,
+  errorCode: string,
+  status = 400,
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/explicit-module-boundary-types
+  data?: any
```
Comment (jflheureux) [inline Website/src/rendering/src/edge/ordercloud/responses.ts:22]: could this be changed to `unknown` type to get rid of the eslint-disable comments?

File: docker-compose.override.yml [hunk 2]
```diff
@@ -53,6 +55,9 @@ services:
       NEXT_PUBLIC_DISCOVER_API_KEY: ${DISCOVER_API_KEY}
       NEXT_PUBLIC_ORDERCLOUD_BUYER_CLIENT_ID: ${ORDERCLOUD_BUYER_CLIENT_ID}
       NEXT_PUBLIC_ORDERCLOUD_BASE_API_URL: ${ORDERCLOUD_BASE_API_URL}
+      ORDERCLOUD_MIDDLEWARE_CLIENT_ID: ${ORDERCLOUD_MIDDLEWARE_CLIENT_ID}
+      ORDERCLOUD_MIDDLEWARE_CLIENT_SECRET: ${ORDERCLOUD_MIDDLEWARE_CLIENT_SECRET}
+      OC_WEBHOOK_HASH_KEY: ${ORDERCLOUD_WEBHOOK_HASH_KEY}
```
Comment (jflheureux) [inline docker-compose.override.yml:60]: This environment variable does not seem to be used. Could it be removed?

File: .env [hunk 3]
```diff
@@ -100,6 +100,9 @@ DISCOVER_API_KEY=
 # OrderCloud
 ORDERCLOUD_BUYER_CLIENT_ID=
 ORDERCLOUD_BASE_API_URL=
+ORDERCLOUD_MIDDLEWARE_CLIENT_ID=
+ORDERCLOUD_MIDDLEWARE_CLIENT_SECRET=
+ORDERCLOUD_WEBHOOK_HASH_KEY=
```
Comment (jflheureux) [inline .env:105]: This environment variable does not seem to be used. Could it be removed?

File: docker-compose.override.yml [hunk 4]
```diff
@@ -39,6 +39,8 @@ services:
       args:
         PARENT_IMAGE: ${REGISTRY}demo/${COMPOSE_PROJECT_NAME}-nodejs:${SITECORE_VERSION}-${NANOSERVER_VERSION}-${DEMO_VERSION}
     mem_limit: 2GB
+    ports:
+      - "8099:3000"
```
Comment (jflheureux) [inline docker-compose.override.yml:43]: Is this port needed for development purposes in general? If it is, could you please add some documentation on how to use it?

File: Website/src/rendering/src/components/Checkout/AddressCard.tsx [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+  if (props.active) {
+    addressClasses.push('address-card__active');
+  }
+
+  const editButton = props.editable && (
+    <button onClick={() => props.onEdit(props.address)}>Edit Address</button>
+  );
+
+  const onClick = (address: DBuyerAddress) => {
+    if (props.onClick && typeof props.onClick === 'function') {
```
Comment (jflheureux) [inline Website/src/rendering/src/components/Checkout/AddressCard.tsx:24]: Is the `typeof function` check necessary? The props already define `onClick` as a function.

File: Website/src/rendering/src/components/Checkout/ShipMethodsList.tsx [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+  selectedShipMethodId?: string;
+  onChange?: (shipMethod: RequiredDeep<ShipMethodSelection>) => void;
+};
+
+const ShipMethodsList = (props: ShipMethodsListProps): JSX.Element => {
+  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
+    const shipMethodId = event.target.value;
+    const shipMethod = props.shipMethods.find((shipMethod) => shipMethod.ID === shipMethodId);
+
+    if (props.onChange && typeof props.onChange === 'function') {
```
Comment (jflheureux) [inline Website/src/rendering/src/components/Checkout/ShipMethodsList.tsx:18]: Is the typeof function check necessary? The props already define onChange as a function.

File: Website/src/rendering/src/components/Forms/AddressForm.tsx [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+      AddressName: addressName,
+      Country: country,
+      Street1: street1,
+      Street2: street2,
+      City: city,
+      State: state,
+      Zip: zip,
+    };
+
+    if (props.onSubmit && typeof props.onSubmit === 'function') {
```
Comment (jflheureux) [inline Website/src/rendering/src/components/Forms/AddressForm.tsx:49]: Is the typeof function check necessary? The props already define onSubmit as a function.

File: Website/src/rendering/src/components/Forms/CreditCardForm.tsx [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+      CardholderName: cardholderName,
+      PartialAccountNumber: cardNumber.slice(-4),
+      ExpirationDate: getIsoDateFromYearAndMonth(expirationYear, expirationMonth),
+      // in a real application we would use a third party payment processor
+      // to securely capture, store, and tokenize sensitive credit card details and store only the opaque token
+      // it is against the terms of service to store sensitive credit card details on the platform
+      Token: '',
+    };
+
+    if (props.onSubmit && typeof props.onSubmit === 'function') {
```
Comment (jflheureux) [inline Website/src/rendering/src/components/Forms/CreditCardForm.tsx:51]: Is the typeof function check necessary? The props already define onSubmit as a function.

File: Website/src/rendering/.babelrc [hunk 9]
```diff
@@ -0,0 +1 @@
+{"presets": ["next/babel"]}
```
Comment (jflheureux) [inline Website/src/rendering/.babelrc:1]: What is this new file allowing us to do?

File: .env (hunk 3 above)
Comment (crhistianramirez) [reply .env:105]: Its actually used by the [ordercloud javascript catalyst](https://github.com/ordercloud-api/ordercloud-javascript-catalyst/blob/main/README.md#usage) library that Oliver built. It is used to authenticate our webhook endpoints and setting it as an environment variable allows that to be done globally.

I could add a comment to make that more clear, what do you think?

File: docker-compose.override.yml (hunk 2 above)
Comment (crhistianramirez) [reply docker-compose.override.yml:60]: Its actually used by the [ordercloud javascript catalyst](https://github.com/ordercloud-api/ordercloud-javascript-catalyst/blob/main/README.md#usage) library that Oliver built. It is used to authenticate our webhook endpoints and setting it as an environment variable allows that to be done globally.

I could add a comment to make that more clear, what do you think?

Comment (jflheureux) [reply docker-compose.override.yml:60]: Yes, a comment in this file would help. Thanks

File: Website/src/rendering/src/components/Checkout/ShipMethodsList.tsx (hunk 6 above)
Comment (crhistianramirez) [reply Website/src/rendering/src/components/Checkout/ShipMethodsList.tsx:18]: The props define it as an optional parameter. Its useful not having to define in storybook for example. If you prefer I can make it required and update our storybook examples though to do a no-op function.

File: .env (hunk 3 above)
Comment (jflheureux) [reply .env:105]: No need for a comment in this file. Just in the `docker-compose.override.yml`. Thanks

File: Website/src/rendering/src/components/Forms/AddressForm.tsx (hunk 7 above)
Comment (crhistianramirez) [reply Website/src/rendering/src/components/Forms/AddressForm.tsx:49]: The props define it as an optional parameter. Its useful not having to define in storybook for example. If you prefer I can make it required and update our storybook examples though to do a no-op function.

File: Website/src/rendering/src/components/Forms/CreditCardForm.tsx (hunk 8 above)
Comment (crhistianramirez) [reply Website/src/rendering/src/components/Forms/CreditCardForm.tsx:52]: The props define it as an optional parameter. Its useful not having to define in storybook for example. If you prefer I can make it required and update our storybook examples though to do a no-op function.

File: Website/src/rendering/.babelrc (hunk 9 above)
Comment (crhistianramirez) [reply Website/src/rendering/.babelrc:1]: This is a workaround to an issue that came up as part of migrating to to nextjs v12 and is documented [here](https://nextjs.org/docs/messages/failed-loading-swc#possible-ways-to-fix-it). I will plan on adding some comments to this file

**Comment by crhistianramirez** [comment]
@jflheureux your PR comments have been addressed. Thank you for reviewing

**Review by jflheureux — APPROVED** [review]

**Pull request merged**